src/main/java/io/anserini/index/IndexRDFCollection.java

```diff
@@ -0,0 +1,224 @@
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+import org.kohsuke.args4j.spi.StringArrayOptionHandler;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Index an RDF Collection, such as Freebase.
+ */
+public class IndexRDFCollection {
+
+  /**
+   * Logger
+   */
+  private static final Logger LOG = LogManager.getLogger(IndexRDFCollection.class);
+
+  public static final class Args {
+    // Required arguments
+
+    @Option(name = "-input", metaVar = "[Directory]", required = true, usage = "collection directory")
+    public String input;
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-collection", required = true, usage = "collection class in io.anserini.collection")
+    public String collectionClass;
+
+    @Option(name = "-generator", required = true, usage = "document generator in io.anserini.index.generator, usually LuceneRDFDocumentGenerator")
+    public String generatorClass;
+
+    // Optional arguments
+
+    @Option(name = "-predicates", handler = StringArrayOptionHandler.class, usage = "predicates to index")
+    public List<String> predicatesToIndex;
+  }
+
+  public final class Counters {
+    public AtomicLong indexedDocuments = new AtomicLong();
+  }
+
+  /**
+   * Default index analyzer (uses StandardAnalyzer)
+   */
+  public final Analyzer ANALYZER;
+
+  /**
+   * Program arguments to hold parameters and properties.
+   */
+  Args args;
+
+  public static void main(String[] args) throws Exception {
+    Args indexRDFCollectionArgs = new Args();
+    CmdLineParser parser = new CmdLineParser(indexRDFCollectionArgs,
+            ParserProperties.defaults().withUsageWidth(90));
+
+    try {
+      parser.parseArgument(args);
+    } catch (CmdLineException e) {
+      System.err.println(e.getMessage());
+      parser.printUsage(System.err);
+      System.err.println("Example command: "+ IndexRDFCollection.class.getSimpleName() +
+              parser.printExample(OptionHandlerFilter.REQUIRED));
+      return;
+    }
+
+    new IndexRDFCollection(indexRDFCollectionArgs).run();
+  }
+
+  private final Path indexPath;
+  private final Path collectionPath;
+  private final Class collectionClass;
+  private final Class transformerClass;
+  private final Collection collection;
+  private final Counters counters;
+
+  /**
+   * Constructor
+   *
+   * @param args program arguments
+   * @throws Exception
+   */
+  public IndexRDFCollection(Args args) throws Exception {
+
+    // Copy arguments
+    this.args = args;
+
+    // Initialize analyzer
+    String defaultAnalyzerName = "org.apache.lucene.analysis.standard.StandardAnalyzer";
```

Comment:
Actually, I'll remove it since I am storing RDF triple components as-is, not analysis is needed for now.